### ISSUE TITLE
Parker convention is not implemented correctly 

### ISSUE THREAD

**Issue by AndreiPashkin** [sanand0/xmljson#40]
It seems like the way of how the library converts XML using Parker convention differs from how it's defined originally in this repository:
https://github.com/doekman/xml2json-xslt

Here is the comparison:

**Input data**
```xml
<root>
    <person>
        <age>12</age>
        <height>1.73</height>
    </person>
    <person>
        <age>12</age>
        <height>1.73</height>
    </person>
</root>
```

**Using xmltodict**

Output:
```json
{
    "person": [
        {
            "age": 12,
            "height": 1.73
        },
        {
            "age": 12,
            "height": 1.73
        }
    ]
}
```

**Using [XSLT document](https://github.com/doekman/xml2json-xslt/blob/master/xml2json.xslt) from the original Google repository**

Output:
```json
{
    "root":[
        {
            "age":12,
            "height":1.73
        },
        {
            "age":12,
            "height":1.73
        }
    ]
}
```

**Comment by dagwieers** [comment]
So **xmljson** (this project) returns this:
```json
{
    "person": [
        {
            "age": 12,
            "height": 1.73
        }, {
            "age": 12,
            "height": 1.73
        }
    ]
}
```
Or this when using `preserve_root=True`:
```json
{
    "root": {
        "person": [
            {
                "age": 12,
                "height": 1.73
            }, {
                "age": 12,
                "height": 1.73
            }
        ]
    }
}
```
To me the XSLT translation seems wrong, it removes information (i.e. the `person` sub-element).
Was there a question ?

**Comment by dagwieers** [comment]
BTW Looking at the examples at https://github.com/doekman/xml2json-xslt/tree/master/unittests your depiction of what **xml2json-xslt** returns is incorrect, it does not turn the root-element into an element named `root`.

**Comment by AndreiPashkin** [comment]
That repository is a reference implementation of Parker convention.

> To me the XSLT translation seems wrong, it removes information (i.e. the `person` element).   

I think it makes perfect sense, because in XML you usually have something like:
```xml
<people>
  <person>
    <name>John</name>
    <age>10</age>
  </person>
</people>
```  
And in JSON you usually don't want to see `{"people": {"person": [...]}}` structure but `{"people": [...]}`.

> your depiction of what **xml2json-xslt** returns is incorrect  

I generated output examples using the XSLT file and `xsltproc` utility.

**Comment by AndreiPashkin** [comment]
Input: https://github.com/doekman/xml2json-xslt/blob/master/unittests/issue1.xml
Output: https://github.com/doekman/xml2json-xslt/blob/master/unittests/issue1.json.expected

Root element seems to be preserved actually

**Comment by dagwieers** [comment]
> Root element seems to be preserved actually

@AndreiPashkin Exactly, so [your example above](https://github.com/sanand0/xmljson/issues/40#issue-414480220) is incorrect. And the output is the same as **xmljson**.

**Comment by dagwieers** [comment]
> And in JSON you usually don't want to see `{"people": {"person": [...]}}` structure but `{"people": [...]}`.

That seems very wrong. Why would someone want to see a sub-element (i.e. `person`) to disappear when doing a conversion? Why the first sub-element ?

**Comment by AndreiPashkin** [comment]
> That seems very wrong. Why would someone want to see a sub-element (i.e. `person`) to disappear when doing a conversion? Why the first sub-element ?

1. Because why would you want to have people -> person structure? It is redundant.
2. Because that's what developers expect from Parker convention implementation.

> @AndreiPashkin Exactly, so [your example above](https://github.com/sanand0/xmljson/issues/40#issue-414480220) is incorrect. And the output is the same as **xmljson**.

Hm, that is correct. I don't why why `xsltproc` worked that way. But that's not important for me anyway, I filed the issue because of the above problem.

**Comment by dagwieers** [comment]
> 1. Because why would you want to have people -> person structure? It is redundant.

The redundancy is in the original data, why would a conversion remove a sub-element layer. That doesn't make sense. A conversion has no semantics.

Maybe you were confused or your examples were wrong and you wanted to preserve/remove the *root-element*. That is possible with **xmljson** using `preserve_root=True/False` as my first example showed.

> I filed the issue because of the above problem.

But there does not seem to be a problem, so can I close this issue ?

**Comment by AndreiPashkin** [comment]
> The redundancy is in the original data, why would a conversion remove a sub-element layer. That doesn't make sense. A conversion has no semantics.

In the original data there is no redundancy. How else would you express array of person-elements in XML? Only with person-tags. But in JSON equivalent of person-tags are dictionaries themselves.

Also - that's how Parker convention defines it. I think it makes sense to implement a standard according to the standard.

**Comment by dagwieers** [comment]
@AndreiPashkin You request is that:
```xml
<people>
  <person>
    <name>John</name>
    <age>10</age>
  </person>
</people>
```
Should be converted to something like this (in JSON):
```xml
<people>
  <name>John</name>
  <age>10</age>
</people>
```
That is what you deemed *that's what developers expect from Parker convention implementation*.
But imagine you have multiple persons like this:
```xml
<people>
  <person>
    <name>John</name>
    <age>10</age>
  </person>
  <person>
    <name>Mary</name>
    <age>12</age>
  </person>
</people>
```
What should that become ?? Because it does not make sense that you get:
```xml
<people>
  <name>John</name>
  <age>10</age>
  <name>Mary</name>
  <age>12</age>
</people>
```
So again, I think you are confused or used the wrong examples and did not understand the difference between *sub-element* and *root-element* in my example. Or you are trolling me. In both cases this ticket should be closed because there is no issue with **xmljson** it works exactly like the XSLT translation you referred to.

**Comment by dagwieers** [comment]
I repeat what I said before, please read it *very carefully*:

> Maybe you were confused or your examples were wrong and you wanted to preserve/remove the _root-element_. That is possible with **xmljson** using `preserve_root=True/False` as my first example showed.

**Comment by javadev** [comment]
```xml
<?xml version="1.0" encoding="UTF-8"?>
<productRating>
  <rating>
    <ratingCode>PG-13</ratingCode>
  </rating>
  <explaination/>
</productRating>
```

May be converted to json
```json
{
  "productRating": {
    "rating": {
      "ratingCode": "PG-13"
    },
    "explaination": {
      "-self-closing": "true"
    }
  }
}
```

https://xmltojson.github.io/json-to-xml.html
